### DISCUSSION TITLE
wire:click seems to only update after I click a second time

### DISCUSSION THREAD

**Discussion by johnrudolph** [livewire/livewire#4509] [1. Help] [answered]
Hello! I am a novice programmer trying to build a simple web game using Laravel and Livewire. I'm calling a function in my Livewire component from a div that uses wire:click. Everything is working as expected, HOWEVER, the view only shows the updated values after I click twice. Looking at the web inspector, the request is going through after the first click. But the view just updates after the second click. Code snippets below. Any ideas? Thank you!!

blade view:
` 

        <div class="a1Right inputarea" wire:click="playTile( {{ $this->a1 }},'right')">
        </div>

        <div class="alSpace space">
            {{ $a1->current_occupant }} 
        </div>

        <div class="b1Space space">
            {{ $b1->current_occupant }} 
        </div>

        <div class="c1Space space">
            {{ $c1->current_occupant }} 
        </div>

        <div class="d1Space space">
            {{ $d1->current_occupant }} 
        </div>`

livewire component:

`<?php

namespace App\Http\Livewire;

use Livewire\Component;
use App\Models\Board;
use App\Models\Space;

class BoardComponent extends Component
{    
    public Board $board;
    public Space $a1;
    public Space $b1;
    public Space $c1;
    public Space $d1;
    public Space $a2;
    public Space $b2;
    public Space $c2;
    public Space $d2;
    public Space $a3;
    public Space $b3;
    public Space $c3;
    public Space $d3;
    public Space $a4;
    public Space $b4;
    public Space $c4;
    public Space $d4;
    public Array $affectedSpaces;

    public function mount()
    {
        // todo: create new board with new game and pass its ID in here.
        $this->board = Board::find(1);
        // todo: seed 16 spaces for a new board, maybe destroy them when game is over?
        $this->a1 = Space::where('position','a1')->where('board_id',$this->board->id)->first();
        $this->a2 = Space::where('position','a2')->where('board_id',$this->board->id)->first();
        $this->a3 = Space::where('position','a3')->where('board_id',$this->board->id)->first();
        $this->a4 = Space::where('position','a4')->where('board_id',$this->board->id)->first();
        $this->b1 = Space::where('position','b1')->where('board_id',$this->board->id)->first();
        $this->b2 = Space::where('position','b2')->where('board_id',$this->board->id)->first();
        $this->b3 = Space::where('position','b3')->where('board_id',$this->board->id)->first();
        $this->b4 = Space::where('position','b4')->where('board_id',$this->board->id)->first();
        $this->c1 = Space::where('position','c1')->where('board_id',$this->board->id)->first();
        $this->c2 = Space::where('position','c2')->where('board_id',$this->board->id)->first();
        $this->c3 = Space::where('position','c3')->where('board_id',$this->board->id)->first();
        $this->c4 = Space::where('position','c4')->where('board_id',$this->board->id)->first();
        $this->d1 = Space::where('position','d1')->where('board_id',$this->board->id)->first();
        $this->d2 = Space::where('position','d2')->where('board_id',$this->board->id)->first();
        $this->d3 = Space::where('position','d3')->where('board_id',$this->board->id)->first();
        $this->d4 = Space::where('position','d4')->where('board_id',$this->board->id)->first();
        $this->affectedSpaces = ['','',''];

    }
    
    public function playTile(Space $space, String $direction)
    {
        $this->affectedSpaces = $this->board->slidingSpaces($space,$direction);

        if(!empty(Space::find($this->affectedSpaces[1])->current_occupant) && !empty(Space::find($this->affectedSpaces[0])->current_occupant) && !empty($space->current_occupant))
        {
            Space::find($this->affectedSpaces[2])
                ->update(['current_occupant' => Space::find($this->affectedSpaces[1])->current_occupant]);
        }
        
        if(!empty(Space::find($this->affectedSpaces[0])->current_occupant) && !empty($space->current_occupant))
        {
            Space::find($this->affectedSpaces[1])
                ->update(['current_occupant' => Space::find($this->affectedSpaces[0])->current_occupant]);
        }

        if(!empty($space->current_occupant))
        {
            Space::find($this->affectedSpaces[0])
                ->update(['current_occupant' => $space->current_occupant]);
        }

        $space->update(['current_occupant' => 'player 1']);
    }

    public function render()
    {

        return view('livewire.board-component', [
            'a1' => $this->a1,
            'a2' => $this->a2,
            'a3' => $this->a3,
            'a4' => $this->a4,
            'b1' => $this->b1,
            'b2' => $this->b2,
            'b3' => $this->b3,
            'b4' => $this->b4,
            'c1' => $this->c1,
            'c2' => $this->c2,
            'c3' => $this->c3,
            'c4' => $this->c4,
            'd1' => $this->d1,
            'd2' => $this->d2,
            'd3' => $this->d3,
            'd4' => $this->d4,
        ]);
    }
}
`

**Comment by joshhanley** [comment, marked as answer]
@johnrudolph it looks like you need to refresh your model from the database after you have updated it, otherwise the latest data won't be available to the view.

Hope this helps!

**Reply by johnrudolph** [reply]
That's exactly it, thanks!! I just added a ->refresh() at the end of the playTile() function and it's working great.

**Reply by joshhanley** [reply]
@johnrudolph no worries! 🙂
